serialization/ksp_plugin.proto

```diff
@@ -0,0 +1,88 @@
+syntax = "proto2";
+
+import "serialization/geometry.proto";
+import "serialization/physics.proto";
+import "serialization/quantities.proto";
+
+package principia.serialization;
+
+message HistoryAndProlongation {
+  required Trajectory history = 1;
+  required Trajectory.Pointer prolongation = 2;
+}
+
+message Vessel {
+  required MasslessBody body = 1;
+  oneof trajectory_bundle {
+    HistoryAndProlongation history_and_prolongation = 2;
+    Trajectory owned_prolongation = 3;
+  }
+}
+
+message Celestial {
+  required MassiveBody body = 1;
+  required HistoryAndProlongation history_and_prolongation = 2;
+}
+
+message Part {
+  required Pair degrees_of_freedom = 1;
+  required Quantity mass = 2;
+  required Multivector gravitational_acceleration_to_be_applied_by_ksp = 3;
+}
+
+message PhysicsBubble {
+  message FullState {
```

Comment:
Do we really need this message?  It could also just be inlined in the containing message.